src/include/kdbtypes.h

```diff
@@ -98,10 +98,12 @@ typedef long double kdb_long_double_t;
 typedef long double kdb_long_double_t;
 #endif
 
-#if SIZEOF_TV_USEC == SIZEOF_INT
-#define ELEKTRA_TIME_USEC_F "%d"
-#else
+#if SIZEOF_TV_USEC == SIZEOF_LONG
 #define ELEKTRA_TIME_USEC_F "%ld"
+#elif SIZEOF_TV_USEC == SIZEOF_LONG_LONG
+#define ELEKTRA_TIME_USEC_F "%lli"
+#else
```

Comment:
>What about also adding == SIZEOF_INT and return an warning if an unhandled size occurs?

i was just wondering: does defining format strings that way even work properly ? for me sizeof(long long) == sizeof(long), and afaik there are systems where sizeof(long) == sizeof(int) and sizeof(int) == sizeof(short) 

what should it default to if an unhandled size occurs ?